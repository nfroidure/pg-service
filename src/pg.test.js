import assert from 'assert';
import { Knifecycle } from 'knifecycle';
import initPGService from './pg';

describe('initPGService', () => {
  const log = jest.fn();
  const ENV = {};
  const PG = {};

  beforeEach(() => {
    log.mockClear();
  });

  test('should work', done => {
    initPGService({
      ENV,
      PG,
      log,
    })
      .then(pg => {
        assert(pg);
        expect(log.mock.calls).toMatchSnapshot();
      })
      .then(() => done())
      .catch(done);
  });

  describe('pg', () => {
    test('should work', done => {
      initPGService({
        ENV,
        PG,
        log,
      })
        .then(pg => {
          assert(pg);
          log.mockClear();

          expect(log.mock.calls).toMatchSnapshot();
        })
        .then(() => done())
        .catch(done);
    });
  });

  test('should work with Knifecycle', done => {
    new Knifecycle()
      .register(initPGService)
      .constant('log', log)
      .run(['pg'])
      .then(({ pg }) => {
        assert(pg);
        expect(log.mock.calls).toMatchSnapshot();
      })
      .then(() => done())
      .catch(done);
  });
});
