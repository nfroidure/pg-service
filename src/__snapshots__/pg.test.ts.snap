// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initPGService pg should work 1`] = `Array []`;

exports[`initPGService should work 1`] = `
Array [
  Array [
    "debug",
    "🐘 - Initializing PG service...",
  ],
]
`;

exports[`initPGService should work with Knifecycle 1`] = `
Array [
  Array [
    "debug",
    "🐘 - Initializing PG service...",
  ],
]
`;
